src/controllers/midi/midicontroller.h

```diff
@@ -1,12 +1,28 @@
 #pragma once
 
+#include <QJSValue>
+#include <utility>
+
 #include "controllers/controller.h"
 #include "controllers/midi/legacymidicontrollermappingfilehandler.h"
 #include "controllers/midi/midimessage.h"
 #include "controllers/softtakeover.h"
 
 class MidiOutputHandler;
 
+class MidiInputHandleJSProxy final : public QObject {
+    Q_OBJECT
+  public:
+    MidiInputHandleJSProxy(
+            const std::shared_ptr<LegacyMidiControllerMapping> mapping,
+            const MidiInputMapping& inputMapping);
+    Q_INVOKABLE bool disconnect();
+
+  protected:
+    std::shared_ptr<LegacyMidiControllerMapping> m_mapping;
+    MidiInputMapping m_inputMapping;
```

Comment:
Is there a risk of memory leak here? I'm not sure whether or not a `MidiInputHandleJSProxy` will ever be collected and in which conditions.

Reply:
So the ownership of the object is basically transferred to the javascript engine and its garbage collector when you call `QJSEngine::newQObject`. So `MidiInputHandleJSProxy` is destroyed once its unreachable from JS and the js-engines garbage collector destroys it. So no risk of memory leaks here. Also a memory leak wouldn't results in crash, just resource exhaustion in the long term.